From a9aaa5e30ed12a1f8deb65d59b31f7b1b4286fad Mon Sep 17 00:00:00 2001
From: jokim-amd <44032736+jokim-amd@users.noreply.github.com>
Date: Fri, 21 Nov 2025 12:53:32 -0500
Subject: [PATCH] hsakmt: bump vgpr count for gfx1151 (#1807)

GFX1151 has 1.5x VGPR memory compared to the rest of GFX11.
---
 projects/rocr-runtime/libhsakmt/src/libhsakmt.h | 1 +
 projects/rocr-runtime/libhsakmt/src/queues.c    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/projects/rocr-runtime/libhsakmt/src/libhsakmt.h b/projects/rocr-runtime/libhsakmt/src/libhsakmt.h
index 4ac445c025c..9581f7bdd43 100644
--- a/projects/rocr-runtime/libhsakmt/src/libhsakmt.h
+++ b/projects/rocr-runtime/libhsakmt/src/libhsakmt.h
@@ -171,6 +171,7 @@ enum full_gfx_versions {
 	GFX_VERSION_YELLOW_CARP	 	= 0x0A0305,
 	GFX_VERSION_PLUM_BONITO		= 0x0B0000,
 	GFX_VERSION_WHEAT_NAS		= 0x0B0001,
+	GFX_VERSION_GFX1151		= 0x0B0501,
 	GFX_VERSION_GFX1200		= 0x0C0000,
 	GFX_VERSION_GFX1201		= 0x0C0001,
 };
diff --git a/projects/rocr-runtime/libhsakmt/src/queues.c b/projects/rocr-runtime/libhsakmt/src/queues.c
index c2a00734ea7..b48235f86f0 100644
--- a/projects/rocr-runtime/libhsakmt/src/queues.c
+++ b/projects/rocr-runtime/libhsakmt/src/queues.c
@@ -116,6 +116,7 @@ uint32_t hsakmt_get_vgpr_size_per_cu(uint32_t gfxv)
 
 	else if (gfxv == GFX_VERSION_PLUM_BONITO ||
 		 gfxv == GFX_VERSION_WHEAT_NAS ||
+		 gfxv == GFX_VERSION_GFX1151 ||
 		 gfxv == GFX_VERSION_GFX1200 ||
 		 gfxv == GFX_VERSION_GFX1201)
 		vgpr_size = 0x60000;
